onnxruntime/core/providers/cpu/optional/optional_ops.cc

```diff
@@ -76,24 +77,26 @@ static void CopySequenceTensor(AllocatorPtr alloc,
   for (; in_tensor != src->end(); ++in_tensor) {
     auto& tensor = in_tensor->Get<Tensor>();
     Tensor tmp(tensor.DataType(), tensor.Shape(), alloc);
-    CopyCpuTensor(&tensor, &tmp);
+    // Using DataTransferManager here allows other non-CPU EPs to use this implementation of the sequence ops
+    (void) data_transfer_mgr.CopyTensor(tensor, tmp);
+
     tgt->Add(std::move(tmp));
   }
 }
 
 static Status PropagateInputOrtValueToFirstOutput(const OrtValue* input_ort_value,
-                                                  OpKernelContext* ctx) {
+                                                  OpKernelContext* ctx,
+                                                  const DataTransferManager& data_transfer_mgr) {
   if (input_ort_value->IsTensor()) {
     const auto* input_tensor = &input_ort_value->Get<Tensor>();
     auto* output_tensor = ctx->Output(0, input_tensor->Shape());
 
     // If the allocation planner had deemed that we re-use the input OrtValue
     // as the output OrtValue, the data pointers in the input_tensor and the
     // output_tensor will be the same and the copy is a no-op.
-    // CopyCpuTensor() already has such copy optimizations - so
+    // DataTransferManager.CopyTensor() already has such copy optimizations - so
```

Comment:
is the comment still valid?

Reply:
It should be, AFAIK CopyTensor should have all needed optimizations.